crates/bevy_state/src/commands.rs

```diff
@@ -13,6 +13,17 @@ pub trait CommandsStatesExt {
     /// Note that commands introduce sync points to the ECS schedule, so modifying `NextState`
     /// directly may be more efficient depending on your use-case.
     fn set_state<S: FreelyMutableState>(&mut self, state: S);
+
+    /// Sets the next state the app should move to, returning a Result.
+    ///
+    /// Returns an error if the [`NextState<S>`](crate::prelude::NextState) resource does not exist.
+    ///
+    /// Internally this schedules a command that updates the [`NextState<S>`](crate::prelude::NextState)
+    /// resource with `state`.
+    ///
+    /// Note that commands introduce sync points to the ECS schedule, so modifying `NextState`
+    /// directly may be more efficient depending on your use-case.
+    fn try_set_state<S: FreelyMutableState>(&mut self, state: S);
```

Comment:
I would prefer to modify set_state to simply return a Result, rather than duplicating the API.